src/asphalt/web/aiohttp.py

```diff
@@ -10,21 +10,24 @@
 from aiohttp.web_response import Response
 from aiohttp.web_runner import AppRunner, TCPSite
 from asphalt.core import (
-    ContainerComponent,
+    Component,
     Context,
-    context_teardown,
+    add_resource,
     resolve_reference,
+    start_service_task,
 )
 
+from ._utils import ensure_server_running
+
 
 @middleware
 async def asphalt_middleware(request: Request, handler: Callable[..., Awaitable]) -> Response:
-    async with Context() as ctx:
-        ctx.add_resource(request, types=[Request])
+    async with Context():
+        add_resource(request, types=[Request])
```

Comment:
```suggestion
        add_resource(request, types=Request)
```

Reply:
Why?

Reply:
`types` accepts a single type, which is the case here, right?

Reply:
No, this is what it says in the docstring:
> :param types: type(s) to register the resource as (omit to use the type of ``value``)

And the annotation is:
> types: type | Sequence[type] = (),

Reply:
hmm so it seems that it can accept a single type, or am I missing something?